### PR TITLE
Setup e2e testing

### PR DESCRIPTION
**Pull request by YasseinBilal**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    ports: [ "8545:8545" ]
+    entrypoint: [ "anvil", "--fork-url", "https://rpc.ankr.com/eth_sepolia", "--host", "0.0.0.0", "--block-time", "0.1", "--silent"] 
+    platform: linux/x86_64
+ 
+  mock-paymaster:
+    image: ghcr.io/pimlicolabs/mock-verifying-paymaster:main
+    ports: [ "3000:3000" ]
+    environment:
+      - ALTO_RPC=http://alto:4337
+      - ANVIL_RPC=http://anvil:8545
```
Comment (kopy-kat) [inline docker-compose.yaml:13]: we could also get rid of the paymaster since we dont use it

File: test/utils/healthCheck.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      if (data.message !== 'pong') {
+        throw new Error('paymaster not ready yet')
+      }
+
+      return
+    } catch {
+      await new Promise((resolve) => setTimeout(resolve, 1000))
+    }
+  }
+}
```
Comment (kopy-kat) [inline test/utils/healthCheck.ts:37]: same as above

File: test/utils/userOps/account.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Address, Hex } from 'viem'
+import { Account } from './types'
+
+export const getAccount = ({
```
Comment (kopy-kat) [inline test/utils/userOps/account.ts:4]: dont we have this function in the sdk?

File: test/utils/userOps/clients.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import {
+  bundlerActions,
+  createBundlerClient,
+  ENTRYPOINT_ADDRESS_V07,
+} from 'permissionless'
+import { pimlicoBundlerActions } from 'permissionless/actions/pimlico'
+import { foundry } from 'viem/chains'
+import { BUNDLER_URL, RPC_URL } from './constants/contracts'
+
+export const getBundlerClient = () =>
```
Comment (kopy-kat) [inline test/utils/userOps/clients.ts:11]: same as above

File: test/utils/userOps/constants/networks.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { Network } from '../types'
+
+export const networks: Network[] = [
+  {
+    name: 'Sepolia',
+    testnet: true,
+    icon: 'https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/info/logo.png',
+    bundlerUrl:
+      'https://api.pimlico.io/v2/sepolia/rpc?apikey=3f08344c-182e-4a2f-adcd-32d1cd057158',
```
Comment (kopy-kat) [inline test/utils/userOps/constants/networks.ts:9]: we should put this in an env file

File: test/utils/userOps/constants/networks.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    rpcUrl: 'https://rpc.ankr.com/eth_sepolia',
+    zeroExBaseUrl: 'https://sepolia.api.0x.org',
```
Comment (kopy-kat) [inline test/utils/userOps/constants/networks.ts:11]: we could also get rid of unused fields here

File: test/utils/userOps/clients.ts (hunk 4 above)
Comment (YasseinBilal) [reply test/utils/userOps/clients.ts:11]: @kopy-kat we don't have this one in the sdk

**Pull request merged**